R/navs-legacy.R

```diff
@@ -105,36 +105,234 @@ navset_hidden <- function(..., id = NULL, selected = NULL,
 #'   character vector, matching the `value` of [nav_panel()]s to be filled, may
 #'   also be provided. Note that, if a `sidebar` is provided, `fillable` makes
 #'   the main content portion fillable.
-#' @param bg a CSS color to use for the navbar's background color.
-#' @param inverse Either `TRUE` for a light text color or `FALSE` for a dark
-#'   text color. If `"auto"` (the default), the best contrast to `bg` is chosen.
+#' @param navbar_options Options to control the appearance and behavior of the
+#'   navbar. Use [navbar_options()] to create the list of options.
+#' @param position `r lifecycle::badge("deprecated")` Please use 
+#'   [`navbar_options = navbar_options(position=)`][navbar_options] instead.
+#' @param collapsible `r lifecycle::badge("deprecated")` Please use 
+#'   [`navbar_options = navbar_options(collapsible=)`][navbar_options] instead.
+#' @param bg `r lifecycle::badge("deprecated")` Please use 
+#'   [`navbar_options = navbar_options(bg=)`][navbar_options] instead.
+#' @param inverse `r lifecycle::badge("deprecated")` Please use 
+#'   [`navbar_options = navbar_options(inverse=)`][navbar_options] instead.
+#' 
 #' @export
 #' @rdname navset
-navset_bar <- function(..., title = NULL, id = NULL, selected = NULL,
-                     sidebar = NULL, fillable = TRUE,
-                     gap = NULL, padding = NULL,
-                     # TODO: add sticky-top as well?
-                     position = c("static-top", "fixed-top", "fixed-bottom"),
-                     header = NULL, footer = NULL,
-                     bg = NULL, inverse = "auto",
-                     collapsible = TRUE, fluid = TRUE) {
+navset_bar <- function(
+  ...,
+  title = NULL,
+  id = NULL,
+  selected = NULL,
+  sidebar = NULL,
+  fillable = TRUE,
+  gap = NULL,
+  padding = NULL,
+  header = NULL,
+  footer = NULL,
+  fluid = TRUE,
+  navbar_options = NULL,
+  position = deprecated(),
+  bg = deprecated(),
+  inverse = deprecated(),
+  collapsible = deprecated()
+) {
   padding <- validateCssPadding(padding)
   gap <- validateCssUnit(gap)
 
-  navs_bar_(
-    ..., title = title, id = id, selected = selected,
-    sidebar = sidebar, fillable = fillable,
-    gap = gap, padding = padding,
+  .navbar_options <- navbar_options_resolve_deprecated(
+    options_user = navbar_options,
     position = position,
-    header = header, footer = footer,
-    bg = bg, inverse = inverse,
-    collapsible = collapsible, fluid = fluid,
+    bg = bg,
+    inverse = inverse,
+    collapsible = collapsible
+  )
+
+  navs_bar_(
+    ...,
+    title = title,
+    id = id,
+    selected = selected,
+    sidebar = sidebar,
+    fillable = fillable,
+    gap = gap,
+    padding = padding,
+    header = header,
+    footer = footer,
+    fluid = fluid,
+    position = .navbar_options$position,
+    bg = .navbar_options$bg,
+    inverse = .navbar_options$inverse,
+    collapsible = .navbar_options$collapsible,
+    underline = .navbar_options$underline %||% FALSE,
```

Comment:
Let's make `underline = TRUE` the global default, making `page_navbar()` and `navset_bar()` consistent.